### PR TITLE
Plug and play plasticity

### PR DESCRIPTION
**Pull request by WilkAndy**

### CODE REVIEW

**Comment by permcody** [comment]
@tonkmr - please review

File: modules/tensor_mechanics/include/materials/ComputeMultiPlasticityStress.h [hunk 1]
```diff
@@ -4,19 +4,19 @@
...
-class FiniteStrainMultiPlasticity;
+class ComputeMultiPlasticityStress;
 
 template<>
-InputParameters validParams<FiniteStrainMultiPlasticity>();
+InputParameters validParams<ComputeMultiPlasticityStress>();
 
 /**
- * FiniteStrainMultiPlasticity is performs the return-map
+ * ComputeMultiPlasticityStress is performs the return-map
```
Comment (jwpeterson) [inline modules/tensor_mechanics/include/materials/ComputeMultiPlasticityStress.h]: "is performs"

File: modules/tensor_mechanics/include/materials/ComputeIncrementalSmallStrain.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "ComputeSmallStrain.h"
+
+/**
+ * ComputeIncrementalSmallStrain defines a strain increment and rotation increment (=1), for small strains.
+ */
+class ComputeIncrementalSmallStrain : public ComputeSmallStrain
+{
+public:
+  ComputeIncrementalSmallStrain(const std:: string & name, InputParameters parameters);
```
Comment (jwpeterson) [inline modules/tensor_mechanics/include/materials/ComputeIncrementalSmallStrain.h]: Space after `std::`

**Comment by WilkAndy** [comment]
I agree with you @jwpeterson about the link to the wiki.  How do other people do it?  In each .h file ?

**Comment by WilkAndy** [comment]
Also lots of the tests are documented too.  I guess we should put a link in the .i files too.

**Comment by jwpeterson** [comment]
> Also lots of the tests are documented too. I guess we should put a link in the .i files too.

This would be my preference, I think... just a one-line comment at the top of the input file with a link to the wiki.  The drawback is that the wiki content could always move, but I think it's still worthwhile.

**Comment by WilkAndy** [comment]
Thanks @jwpeterson ,

a

**Pull request merged**